webapp/channels/src/components/channel_notifications_modal/channel_notifications_modal.tsx

```diff
@@ -36,508 +42,296 @@ type Props = PropsFromRedux & {
     currentUser: UserProfile;
 };
 
-export type ChannelMemberNotifyProps = Partial<ChannelNotifyProps> & Pick<UserNotifyProps, 'desktop_threads' | 'push_threads'>
-
-type State = {
-    show: boolean;
-    activeSection: string;
-    serverError: string | null;
-    desktopNotifyLevel: ChannelNotifyProps['desktop'];
-    desktopSound: ChannelNotifyProps['desktop_sound'];
-    desktopNotifySound: ChannelNotifyProps['desktop_notification_sound'];
-    desktopThreadsNotifyLevel: UserNotifyProps['desktop_threads'];
-    markUnreadNotifyLevel: ChannelNotifyProps['mark_unread'];
-    pushNotifyLevel: ChannelNotifyProps['push'];
-    pushThreadsNotifyLevel: UserNotifyProps['push_threads'];
-    ignoreChannelMentions: ChannelNotifyProps['ignore_channel_mentions'];
-    channelAutoFollowThreads: ChannelNotifyProps['channel_auto_follow_threads'];
-};
-
-export type DesktopNotificationProps = Pick<State, 'desktopNotifyLevel' | 'desktopNotifySound' | 'desktopSound' | 'desktopThreadsNotifyLevel'>
-export type PushNotificationProps = Pick<State, 'pushNotifyLevel' | 'pushThreadsNotifyLevel'>
+function getUseSameDesktopSetting(currentUserNotifyProps: UserNotifyProps, channelMemberNotifyProps?: ChannelMemberNotifyProps) {
```

Comment:
Could we re-write this to just make use of `if` statements instead? I think it would be much clearer and easier to read.